snuba/state/stateful_count.py

```diff
@@ -0,0 +1,60 @@
+from threading import Thread
+from time import time
+from typing import Any, Sequence, Tuple
+
+from arroyo.processing.strategies.dead_letter_queue import CountInvalidMessagePolicy
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    InvalidMessage,
+)
+from arroyo.types import TPayload
+
+from snuba.redis import redis_client
+
+
+class StatefulCountInvalidMessagePolicy(CountInvalidMessagePolicy[TPayload]):
+    """
+    An extension of the CountInvalidMessagePolicy which is able to save and load
+    the state of counted hits in Redis
+    """
+
+    def __init__(self, redis_hash_name: str, limit: int, seconds: int = 60) -> None:
+        self.__name = redis_hash_name
```

Comment:
just curious is the `redis_hash_name` going to be set by the consumer? like will different consumers have different hash names?

Reply:
Yeah each consumer would have its own DLQ, which means a policy would be instantiated per consumer. The name of the hash storing the counts would be set wherever the consumer is instantiated with a DLQ.